From 4161593c24ea24f03bc89d74147e7209dc02ba80 Mon Sep 17 00:00:00 2001
From: Mike Yuan <me@yhndnzj.com>
Date: Sat, 3 Dec 2022 20:27:47 +0800
Subject: [PATCH] rpm/systemd-update-helper: use --no-warn when disabling units

Suppress the "empty [Install] section" warning (see #25437).

(cherry picked from commit 0acb1459a15f5b4d3a9bd2e7bf52661ca7bdebf0)

Related: #2141979
---
 src/rpm/systemd-update-helper.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/rpm/systemd-update-helper.in b/src/rpm/systemd-update-helper.in
index ab8cdc0ff9..b35d952fab 100755
--- a/src/rpm/systemd-update-helper.in
+++ b/src/rpm/systemd-update-helper.in
@@ -19,21 +19,21 @@ case "$command" in
 
     remove-system-units)
         if [ -d /run/systemd/system ]; then
-            systemctl --no-reload disable --now "$@"
+            systemctl --no-reload disable --now --no-warn "$@"
         else
-            systemctl --no-reload disable "$@"
+            systemctl --no-reload disable --no-warn "$@"
         fi
         ;;
 
     remove-user-units)
-        systemctl --global disable "$@"
+        systemctl --global disable --no-warn "$@"
 
         [ -d /run/systemd/system ] || exit 0
 
         users=$(systemctl list-units 'user@*' --legend=no | sed -n -r 's/.*user@([0-9]+).service.*/\1/p')
         for user in $users; do
             SYSTEMD_BUS_TIMEOUT={{UPDATE_HELPER_USER_TIMEOUT}} \
-                    systemctl --user -M "$user@" disable --now "$@" &
+                    systemctl --user -M "$user@" disable --now --no-warn "$@" &
         done
         wait
         ;;
